server/requestorserver/server.go

```diff
@@ -455,16 +484,19 @@ func (s *Server) handlePublicKey(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) doResultCallback(result *server.SessionResult) {
-	url := s.irmaserv.GetRequest(result.Token).Base().CallbackURL
-	if url == "" {
-		return
-	}
-	server.DoResultCallback(url,
-		result,
-		s.conf.JwtIssuer,
-		s.irmaserv.GetRequest(result.Token).Base().ResultJwtValidity,
-		s.conf.JwtRSAPrivateKey,
-	)
+	if request, err := s.irmaserv.GetRequest(result.Token); err == nil {
+		url := request.Base().CallbackURL
+		if url == "" {
+			return
+		}
+		server.DoResultCallback(url,
+			result,
+			s.conf.JwtIssuer,
+			request.Base().ResultJwtValidity,
+			s.conf.JwtRSAPrivateKey,
+		)
+	}
+	return
```

Comment:
You can also do an early return here, so on line 487 you can do 
```
request, err := s.irmaserv.GetRequest(result.Token)
if err != nil {
    return nil
}
...
```
Then you don't have the extra intent.

Isn't it useful to log this error, since it is not expected to happen?

Reply:
changed the order as suggested. the error already gets logged when it occurs further down in the code

Reply:
The error returned by `s.sessions.get(token)` in `GetRequest` seems not to be logged. Or do I miss something there?

Reply:
When I check the code, I cannot find any flow where the error is not logged. In case of the memory store, no error is produced in the get request. In case of the redis store, any error is logged with the `logAsRedisError(err)` function in `sessions.go` before it is returned and sent back to `GetRequest`, so logging it again in `GetRequest` would only duplicate it.

Have you spot a case where this is not the case?